bin/add-exercise

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if (( $# != 1 )); then
+    echo >&2 "Populates a new directory for a practice exercise."
+    echo >&2 "Usage: $0 <exercise-slug>"
+    exit 1
+fi
+
+die() { echo >&2 "$*"; exit 1; }
+
+required_tool() {
+    command -v "$1" >/dev/null 2>&1 ||
+        die "$1 is required but not installed. Please install it and make sure it's in your PATH."
+}
+required_tool jq
+required_tool curl
+
+[[ -f ./bin/fetch-configlet ]] || die "run this script from the repo's root directory."
+
+slug="${1}"
+name=$(echo "${slug}" | sed 's/\b\w/\u&/g; s/-/ /g')    # assumes GNU sed
+
+it_exists=$(
+    jq --arg slug "${slug}" '
+        .exercises.practice
+        | map(select(.slug == $slug))
+        | length > 0
+    ' config.json
+)
+[[ ${it_exists} == false ]] || die "${slug} already exists in config.json"
+
+# Add entry for exercise in config.json
+./bin/fetch-configlet
+jq  --arg slug "${slug}"                 \
+    --arg uuid "$(./bin/configlet uuid)" \
+    --arg name "${name}"                 \
+    '
+        .exercises.practice += [
+            {
+                slug: $slug,
+                name: $name,
+                uuid: $uuid,
+                practices: [],
+                prerequisites: [],
+                difficulty: 1
+            }
+        ]
+    ' config.json > config.json.tmp \
+&& mv config.json.tmp config.json
+
+# Sync the exercise
+./bin/configlet sync --update --yes  \
+                     --tests include \
+                     --metadata      \
+                     --docs          \
+                     --exercise "${slug}"
+
+touch "exercises/practice/${slug}/.meta/example.jq"
+
+cat <<END_SLUG > "exercises/practice/${slug}/${slug}.jq"
```

Comment:
Would an `echo` or `printf` work here?

Reply:
Yes. It results in a long line, but c'est la vie.